Hi @AzureTriple are you actually testing this code before your submit your PR? The reason I ask is as follows
* this build is totally broken....
* for example, in [settings.gradle.kts](https://github.com/AzureTriple/nutch/blob/NUTCH-2940/settings.gradle.kts#L28-L30) you are including plugins in the build which cannot even be built! This fails the build immediately.
* If I comment out those lines, clean the code base, I still get the following

```
* What went wrong:
Execution failed for task ':job'.
> Cannot expand ZIP '/Users/lmcgibbn/Downloads/nutch/build/classes' as it is not a file.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 732ms
4 actionable tasks: 3 executed, 1 up-to-date
```

Please make sure you are testing the gradle task functionality before you submit your PR.

Also, rather than retaining the default message when you open your pull request, please describe what work you have actually done. This really helps out your fellow developers. Thank you 👍

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -303,29 +303,31 @@ tasks.register<Copy>("runtime")
 
 tasks.register<Jar>("job")
 {
+    //TODO there is no support to create a ".job" directly
     description = "Make nutch.job jar"
     dependsOn("compile")
+
     from(
-        zipTree("${project.properties["build.classes"]}").matching {
+        files("${project.properties["build.classes"]}") {
             exclude("nutch-default.xml","nutch-site.xml")
         },
-        zipTree("${project.properties["conf.dir"]}").matching {
+        files("${project.properties["conf.dir"]}") {
             exclude("*.template","hadoop*.*")
         },
-        zipTree("${project.properties["build.lib.dir"]}").matching {
+        files("${project.properties["build.lib.dir"]}") {
             eachFile {
                 relativePath = RelativePath(true,"lib")
             }
             include("**/*.jar")
             exclude("hadoop-*.jar,slf4j*.jar","log4j*.jar")
         },
-        zipTree("${project.properties["build.plugins"]}").matching {
+        files("${project.properties["build.plugins"]}") {
             eachFile {
                 relativePath = RelativePath(true,"classes","plugins")
             }
         }
     )
-    into(layout.buildDirectory.dir("${project.properties["build.dir"]}/${project.properties["final.name"]}.job"))
+    into("${project.properties["build.dir"]}/${project.properties["final.name"]}.job")
 }
 
 tasks.register<JavaCompile>("compile-core-test")

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -25,6 +25,8 @@
  */
 
 rootProject.name = "nutch"
+/* TODO uncomment when plugins are completed
 includeBuild("src/plugin/sample-plugin")
 includeBuild("src/plugin/tld")
 includeBuild("src/plugin/language-identifier")
+*/

